pallets/asset/src/lib.rs

```diff
@@ -1627,7 +1636,9 @@ impl<T: Trait> Module<T> {
         Self::ensure_asset_fresh(&ticker)?;
         let ticker_details = Self::ticker_registration(&ticker);
 
-        <Identity<T>>::consume_auth(ticker_details.owner, Signatory::from(to_did), auth_id)?;
+        let signer = Signatory::from(to_did);
```

Comment:
I'd like to avoid introducing such a function to force us to think about the transactionality aspect instead. `try_and_consume` is basically `consume_auth` but renamed. It's still an attractive nuisance basically.